dandi/dandiapi.py

```diff
@@ -77,6 +77,17 @@ class VersionStatus(Enum):
     PUBLISHED = "Published"
 
 
+def get_api_key_env_name(dandi_instance: DandiInstance) -> str:
+    """
+    Get the name of the environment variable that can be used to specify the
+    API key for the given DANDI instance.
+
+    :param dandi_instance: the DANDI instance
+    :return: the name of the environment variable
+    """
+    return f"{dandi_instance.name.upper().replace('-', '_')}_API_KEY"
```

Comment:
looking at multitude of calls , I wonder if we should add `api_key_envvar` property to `dandi_instance`'s class and be done to use it -- that would make code shorter.